lib/src/reporting_query_parser.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:convert';
+
+class ReportingQueryParser {
+  static const _emptyEntry = '';
+
+  /// Parse queryId from a json string.
+  static String parseQueryIdFromJsonString(String jsonString) =>
+      json.decode(jsonString)['queryId'] ?? _emptyEntry;
+
+  /// Parse google storage download from a json string.
+  static String parseDownloadPathFromJsonString(String jsonString) =>
+      (json.decode(jsonString)['metadata'] ??
+          const {})['googleCloudStoragePathForLatestReport'] ??
+      _emptyEntry;
+
+  /// Parse revenue from a json string.
+  static Map<String, String> parseRevenueFromString(String responseFull) {
+    String responseBody =
```

Comment:
Let me know if this is readable? The intent is do a nested null operator on Maps :(